packages/commercetools/theme/components/Checkout/UserBillingAddresses.vue

```diff
@@ -42,32 +42,34 @@ import {
   SfCheckbox,
   SfAddressPicker
 } from '@storefront-ui/vue';
-import { userBillingGetters } from '@vue-storefront/commercetools';
-
+import { useUserBilling, userBillingGetters } from '@vue-storefront/commercetools';
 export default {
   name: 'UserBillingAddresses',
   props: {
     currentAddressId: {
-      type: Number,
+      type: String | Number,
       required: true
     },
     setAsDefault: {
       type: Boolean,
       required: true
-    },
-    billingAddresses: {
-      type: Array,
-      required: true
     }
   },
   components: {
     SfCheckbox,
     SfAddressPicker
   },
   setup (_, { emit }) {
-    const setCurrentAddress = $event => emit('setCurrentAddress', $event);
-
+    const { billing: userBilling } = useUserBilling();
+    const setCurrentAddress = async addressId => {
+      const selectedAddress = userBillingGetters.getAddresses(userBilling.value, { id: addressId });
+      if (!selectedAddress || !selectedAddress.length) {
```

Comment:
do we need both?

Reply:
There is possibility that selectedAdddress is null or undefined:
```js
getAddresses: (shipping, criteria?: Record<string, any>) => {
    if (!criteria || !Object.keys(criteria).length) {
      return shipping.addresses;
    }

    const entries = Object.entries(criteria);
    return shipping.addresses.filter(address => entries.every(([key, value]) => address[key] === value));
  },
```